astropy/units/decorators.py

```diff
@@ -6,13 +6,16 @@
 from numbers import Number
 from collections.abc import Sequence
 import inspect
+import typing as T
 
 import numpy as np
 
 from astropy.utils.decorators import wraps
 from .core import (Unit, UnitBase, UnitsError, add_enabled_equivalencies,
                    dimensionless_unscaled)
 from .physical import _unit_physical_mapping
+from .quantity import Quantity
+from ._typing import _isAnnotated, _isUnion, NoneType
```

Comment:
If we're importing them from another module, the names should not be private.

Reply:
If I move typing to extern, does that still apply?